src/ConfigEditor.tsx

```diff
@@ -22,16 +23,24 @@ export class ConfigEditor extends PureComponent<Props, State> {
     // Secure field (only sent to the backend)
     onAPIKeyChange = (event: ChangeEvent<HTMLInputElement>) => {
         const { onOptionsChange, options } = this.props;
-        const jsonData = {
-            ...options.jsonData,
+        if (options.jsonData.apiKey) {
+            // special backwards compatibility thing...
+            // the old api key was stored unencrypted, meaning it'll be set to empty string and from now on only the encrypted property will be used...
+            console.info("the api key was stored unencrypted... clearing it from the unencrypted store and storing it encrypted...");
+            // not sure if this is the correct way to do these things but calling onOptionsChange(...) twice in a row in short succession leads to some janky behaviour...
+            options.jsonData.apiKey = '';
```

Comment:
where is it converted from an unsecure to a secure object?

Reply:
just a few lines below :wink: where it uses `secureJsonData` instead of `jsonData` to change the value. Not sure if I should do an empty check and use the unencrypted property instead of the user input value :thinking:

Reply:
it's not transferred per-se, rather just set to an empty string. the secure property is then used to hold the new value, hence my hint at maybe solving this differently (extracting the unsecured value to the secure object) ^^

I already tried setting it to null, maybe i need to modify some other parts aswell ^^

Reply:
@mariusgiger "migrating" the property from unsecured to secured doesn't work as initially thought, as this will only execute on value change... hence the new api key needs to be copy/pasted or manually entered (but who in their right state of mind would do that?!) ^^

Reply:
the thing is that '' will result in true, therefore this will always be invoked, correct?

ok, I think this is not something that will happen..maybe we need to add a hint?

Reply:
nope, empty strings, the number 0, empty arrays or objects and many more things are always false in JS/TS... But I changed it to null now anyhow :wink: